src/ddci.c

```diff
@@ -707,9 +707,10 @@ int ddci_create_pmt(ddci_device_t *d, SPMT *pmt, uint8_t *new_pmt, int pmt_size,
 
     // Add Stream pids
     // Add CA IDs and CA Pids
+    adapter *ddci_adapter = get_adapter(d->id);
+    
     for (i = 0; i < pmt->stream_pids; i++) {
-
-        if (get_ca_multiple_pmt(pmt->adapter)) {
+        if (get_ca_multiple_pmt(ddci_adapter->id)) {
```

Comment:
This is d->id

Reply:
Okay, so the call to get_adapter() isn't needed at all?

Reply:
Yeah

Reply:
Could be that I'll be able to drop the changes to test_ddci.c too then

Reply:
pmt->adapter is different than d->id and I think the change to d->id is correct.

Reply:
Compiles, but haven't tested it